xmipptomo/protocols/protocol_subtomo_map_back.py

```diff
@@ -88,166 +103,269 @@ def _defineParams(self, form):
-                    coord = subtomo.getCoordinate3D()
-                    coord.setVolume(tomo)
-                    nRow.setValue(lib.MDL_XCOOR, int(coord.getX(const.BOTTOM_LEFT_CORNER)*scaleFactor))
-                    nRow.setValue(lib.MDL_YCOOR, int(coord.getY(const.BOTTOM_LEFT_CORNER)*scaleFactor))
-                    nRow.setValue(lib.MDL_ZCOOR, int(coord.getZ(const.BOTTOM_LEFT_CORNER)*scaleFactor))
-                    # Compute inverse matrix
-                    #A = subtomo.getTransform().getMatrix()
-                    #subtomo.getTransform().setMatrix(np.linalg.inv(A))
-                    # Convert transform matrix to Euler Angles (rot, tilt, psi)
-                    from pwem import ALIGN_PROJ
-                    alignmentToRow(subtomo.getTransform(convention="xmipp"), nRow, ALIGN_PROJ)
-                    nRow.addToMd(mdGeometry)
-            fnGeometry = self._getExtraPath("geometry%d.xmd" % classId)
-            mdGeometry.write(fnGeometry)
-
-            if TsSubtomo != TsTomo:
-                factor = TsSubtomo/TsTomo
-                args = "-i %s -o %s --scale %d" % (ref, ref, factor)
-                self.runJob('xmipp_transform_geometry', args)
-
-            if self.paintingType.get() == 0:
-                painting = 'copy'
-            elif self.paintingType.get() == 1:
-                painting = 'avg %d' % self.threshold.get()
-            elif self.paintingType.get() == 2:
-                painting = 'highlight %d' % self.constant.get()
-            elif self.paintingType.get() == 3:
-                painting = 'copy_binary %f' % self.threshold.get()
-
-            tomogram = self._getExtraPath("tomogram_%d.mrc" % tomo.getObjId())
-            args = " -i %s -o %s --geom %s --ref %s --method %s" % (tomogram, tomogram,
-                                                                    self._getExtraPath("geometry%d.xmd" % classId),
-                                                                    ref, painting)
-            self.runJob("xmipp_tomo_map_back", args)
+                self.tomos = input.getPrecedentsInvolved()
+
+        return self.tomos
+    def useOtherSetOfTomograms(self):
+        """ Returns true if tomograms to be used are those in inputTomograms"""
+        return self.inputTomograms.get() is not None
+
+    def getInputSetOfTomograms(self):
+        """ Returns the set of tomograms form the input"""
+
+        if self.useOtherSetOfTomograms():
+            return self.inputTomograms.get()
+        else:
+            input = self._getInput()
+
+            if isinstance(input, SetOfSubTomograms):
+                input = input.getCoordinates3D()
+
+            return input.getPrecedents()
+
+    def _getInput(self):
+        """ Returns the iterator on the input set that could be on classes, subtomograms ot coordinates"""
+        if self._useClasses():
+            return self.inputClasses.get().getFirstItem()
+        else:
+            return self.inputSubtomos.get()
```

Comment:
and if it is a set of coordinates?

Reply:
inputSubtomos is miss-leading. The same param will contain either subtomograms or coordinates. I didn't want to rename the parameter and break betatesting projects.